lib/internal/util.js

```diff
@@ -36,6 +37,35 @@ function objectToString(o) {
 // each one once.
 const codesWarned = {};
 
+// Returns a function that emits a deprecation on first run
+// For internal use only. `code` is a required argument.
+// If --no-deprecation is set, then it is a no-op.
+// { pending: true } are visible only when pendingDeprecation is on
+function deprecator(msg, code, options = {}) {
+  if (process.noDeprecation === true) {
+    return () => {};
+  }
+  if (options.pending && !pendingDeprecation) {
+    return () => {};
+  }
+
+  // code should be set
+  if (typeof code !== 'string' || code === '')
```

Comment:
I personally do not feel that the `code === ''` check is really necessary but it would actually be nice to check for the right pattern as in: `!/^DEP(\d\d\d\d|00XX|0XXX|XXXX)$/.test(code)`.

That aside: it is in both cases not a `ERR_INVALID_ARG_TYPE`. It could be a `ERR_INVALID_ARG_VALUE`.

Reply:
I don’ think it’s needed to validate the pattern.

+1 on ERR_INVALID_ARG_VALUE